### ISSUE TITLE
[FR] Full --unattended support

### ISSUE THREAD

**Issue by DerDanilo** [pivpn/pivpn#1275]
**Is your feature request related to a problem? Please describe.**
- The script requires a frontend and cannot be run without one unless `DEBIAN_FRONTEND=noninteractive` is provided.
- Not all important configurations are possible to set via `setupVARS.conf` (see https://github.com/pivpn/pivpn/issues/1273)

**Describe the solution you'd like**
- Support additional flag e.g. `--nofrontend` that disables the frontend parts. Usually one doesn't care about progress output when running in unattended mode.
- Read all config items from `setupVARS.conf` (see https://github.com/pivpn/pivpn/issues/1273)

**Describe alternatives you've considered**
- Mention in the manual that `DEBIAN_FRONTEND=noninteractive` needs to be defined when frontend is not given.
- Ansible editing downloaded setup script, replacing desirged config lines (which might not work with a script update)

**Additional context**
I discovered this problem while writing an Ansible Playbook to deploy pivpn + pivpn-webui on a Debian VM.

**Comment by orazioedoardo** [comment]
> Support additional flag e.g. --nofrontend that disables the frontend parts. Usually one doesn't care about progress output when running in unattended mode.

There is no progress bar when using `--unattended`

**Comment by DerDanilo** [comment]
> There is no progress bar when using `--unattended`

When using `--unattended` I still see progress bars and package installation "GUI" on CLI. Hence the installation without any frontend fails. I tested this on multiple systems.

**Comment by orazioedoardo** [comment]
> When using --unattended I still see progress bars and package installation "GUI" on CLI.

How are you providing the unattended flag? https://docs.pivpn.io/install/#non-interactive-installation

**Comment by DerDanilo** [comment]
requiring frontend
`/bin/bash /root/installpivpn.sh --unattended /root/setupvars.conf`

not requiring frontend
`/bin/bash DEBIAN_FRONTEND=noninteractive /root/installpivpn.sh --unattended /root/setupvars.conf`

**Comment by 4s3ti** [comment]
@DerDanilo We are using CI scripts to test pivpn and it is working just fine ... it feels there are some details missing ... can't really understand what

**Comment by DerDanilo** [comment]
This is the part of the Ansible role that executes the script. Currently there is no checksum verification though.

Without `DEBIAN_FRONTEND=noninteractive` it does not work. I also tried the Ansible `command` module instead of `shell` module , no difference.

```
- name: "Init install tasks"
  become: yes
  block:
    - name: "Create script directory"
      file:
        path: "{{ wireguard_pivpn_script_path }}"
        state: directory
        owner: "{{ wireguard_pivpn_script_owner }}"
        group: "{{ wireguard_pivpn_script_owner }}"
        mode: 0755

    - name: "Deploy setupvars.conf"
      template:
        src: "setupvars.conf.j2"
        dest: "{{ wireguard_pivpn_setup_vars_file }}"
        owner: "{{ wireguard_pivpn_script_owner }}"
        group: "{{ wireguard_pivpn_script_owner }}"
        mode: "a+x"

    - name: "Download pivpn setup script"
      get_url:
        url: "{{ wireguard_pivpn_setup_script_url }}"
        dest: "{{ wireguard_pivpn_script }}"
        owner: "{{ wireguard_pivpn_script_owner }}"
        group: "{{ wireguard_pivpn_script_owner }}"
        mode: 'a+x'
      register: _download_script
      until: _download_script is succeeded
      retries: 5
      delay: 5
      check_mode: false

    # Edit pivpn setup script
    - name: "Edit setup script wireguard subnet"
      lineinfile:
        path: "{{ wireguard_pivpn_script }}"
        state: present
        regexp: '(.*?)pivpnNET="10.6.0.0"(.*)'
        line: '   pivpnNET="{{ wireguard_pivpn_pivpnnet }}"'

    - name: "Edit setup script wireguard allowed ips"
      lineinfile:
        path: "{{ wireguard_pivpn_script }}"
        state: present
        regexp: '(.*?)ALLOWED_IPS="0.0.0.0/0(.*)'
        line: '   ALLOWED_IPS="{{ wireguard_pivpn_allowed_ips | default(wireguard_pivpn_allowed_ips_default) }}"'

    - debug:
        msg: "Running pivpn script now, this may take a while..."

    - debug:
        msg: "/bin/bash DEBIAN_FRONTEND=noninteractive {{ wireguard_pivpn_script }} --unattended {{ wireguard_pivpn_setup_vars_file }}"

    - name: "Run pivpn setup script"
      shell: 
        cmd: 'DEBIAN_FRONTEND=noninteractive {{ wireguard_pivpn_script }} --unattended {{ wireguard_pivpn_setup_vars_file }}'
        chdir: "{{ wireguard_pivpn_script_path }}"
      args:
        executable: /bin/bash
      register: __script_output

    - debug:
        var: __script_output.stdout_lines

  when: wireguard_pivpn_init_setup|bool
  tags: ["wireguard_pivpn"]
```

**Comment by 4s3ti** [comment]
> requiring frontend
> `/bin/bash /root/installpivpn.sh --unattended /root/setupvars.conf`
> 
> not requiring frontend
> `/bin/bash DEBIAN_FRONTEND=noninteractive /root/installpivpn.sh --unattended /root/setupvars.conf`

hmmm .... `installpivpn.sh`  what is that one doing?

**Comment by DerDanilo** [comment]
> hmmm .... `installpivpn.sh` what is that one doing? that is definitely not our scripts

I assumed that it was clear. My bad. This is of course the installer script (`install.sh`) that is downloaded into a script not just called "`install.sh`" since this could conflict with other scripts.

Either there is something wrong with the installer script not silencing all UI options properly; or something in Ansible. Currently I rather assume that there is something with the script since Ansible complained already (by doing nothing) that the script won't actually work. (Without `DEBIAN_FRONTEND=noninteractive` set)

Can someone please try to run the current installer script with the --unattended option and check regarding the progress bars still shown? Thanks

**Comment by orazioedoardo** [comment]
> There is no progress bar when using `--unattended`

Well, I recalled wrong. Unattended only "no input", not "no input and textual only".

**Comment by 4s3ti** [comment]
this seems to be fixed with #1276 which was merged to master. closing.
